### PR TITLE
Controllers + API endpoints

### PR DESCRIPTION
**Pull request by hachiyuki8**
- All endpoints have to be prefixed with `api/`
- `/slips`: return all slips, with optional query parameters:
    - `before` and `after`: filter by timestamp (inclusive bounds), argument has to be an epoch time
    - `driver_id`: filter by driver id
    - `company_id`: filter by company id associated with the driver id stored
    - example: `/slips?driver_id=d1&company_id=c1&before=1612000000000`
- `/slips/:id`: return the slip with the corresponding id, or return 404 if not found 
- `/drivers`: return all drivers, with optional query parameters:
    - `company_id`: filter by company id
- `/drivers/:id`: return the driver with the corresponding id, or return 404 if not found

- Also enforced unique ID constraints for all the models and referential integrity for any cross references

### CODE REVIEW

File: server/src/controllers/slips.js [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          return res.status(200).json(slips)
+        }
+      })
+      .catch(err => {
+        console.log("getSlips: " + err)
+        return res.status(500).json(err)
+      });
+  };
+    
+  getSlipById = async (req, res) => {
```
Comment (meghrajeev) [inline server/src/controllers/slips.js:36]: Could you please tell me where the endpoint/ controller to fetch slips based on a date range is? Also, to fetch drivers from drivers table using id.

Comment (hachiyuki8) [reply server/src/controllers/slips.js:36]: the endpoints and corresponding behaviors are listed in the PR description above
backend is not currently deployed yet, so the only way to actually hit it is to run the server locally

**Comment by meghrajeev** [comment]
Yes yes, that I understand. But just wanted to know if we also have one for fetching from drivers using id also.
Will approve this for now

**Review by meghrajeev — APPROVED** [review]

**Comment by hachiyuki8** [comment]
> Yes yes, that I understand. But just wanted to know if we also have one for fetching from drivers using id also. Will approve this for now

Oh I see. Yes, I will add that as well :)

**Pull request merged**